primitives/arithmetic/src/traits.rs

```diff
@@ -79,6 +79,15 @@ pub trait AtLeast32Bit: BaseArithmetic + From<u16> + From<u32> {}
 
 impl<T: BaseArithmetic + From<u16> + From<u32>> AtLeast32Bit for T {}
 
+/// A meta trait for arithmetic.
+///
+/// Arithmetic types do all the usual stuff you'd expect numbers to do. They are guaranteed to
+/// be able to represent at least `u32` values without loss, hence the trait implies `From<u32>`
+/// and smaller integers. All other conversions are fallible. And is unsigned.
```

Comment:
I'd be fine here by just saying: 

```suggestion
/// Same as [`AtLeast32Bit `], but also bounded to be unsigned. 
```